hack/local-up-cluster.sh

```diff
@@ -521,43 +550,61 @@ EOF
 
 # start_discovery relies on certificates created by start_apiserver
 function start_discovery {
-    # TODO generate serving certificates
+    curl --silent -k -g $API_HOST:31090
+    if [ ! $? -eq 0 ]; then
+        echo "Kubernetes Discovery secure port is free, proceeding..."
+    else
+        echo "ERROR starting Kubernetes Discovery, exiting. Some process on $API_HOST is serving already on $DISCOVERY_SECURE_PORT"
+        return
+    fi
+
+     # Create serving ca
+    sudo /bin/bash -e <<EOF
+    rm -f "${CERT_DIR}/serving-ca.crt" "${CERT_DIR}/serving-ca.key"
+    openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout "${CERT_DIR}/serving-ca.key" -out "${CERT_DIR}/serving-ca.crt" -subj "/C=xx/ST=x/L=x/O=x/OU=x/CN=ca/emailAddress=x/"
+    echo '{"signing":{"default":{"expiry":"43800h","usages":["signing","key encipherment","server auth","client auth"]}}}' > "${CERT_DIR}/serving-ca-config.json"
+EOF
+    create_serving_certkey serving-ca etcd etcd.kube-public.svc
+    create_serving_certkey serving-ca kubernetes-discovery api.kube-public.svc "localhost" ${API_HOST_IP}
+    # etcd doesn't seem to have separate signers for serving and client trust
+    create_client_certkey serving-ca discovery-etcd discovery-etcd
+
     create_client_certkey client-ca discovery-auth system:discovery-auth
     write_client_kubeconfig discovery-auth
+    ${GO_OUT}/kubectl --kubeconfig="${CERT_DIR}/discovery-auth.kubeconfig" config set-cluster local-up-cluster --insecure-skip-tls-verify --server="https://kubernetes.default.svc"
```

Comment:
I'd like to get the connections working and then I'll come back for certs.  I suspect I'll refactor genericapiserver to work with in-cluster config.